#8068: Remove spurious docs jobs / workflows / executions

dimitri-tenstorrent opened the pull request:
Resolve #8068 
Resolve #8097 

@TT-billteng also pushed me to get this https://github.com/tenstorrent/tt-metal/issues/8097 over the line :) 

Moved `docs-latest-public.yaml` into `all-post-commit-workflows.yaml`'

Removed the doc step from `fast-dispatch-build-and-unit-tests.yaml` and also from the secret steps hidden within `slow and fast dispatch tests (`run_pre_post_commit_regressions_fast_dispatch.sh` and `run_pre_post_commit_regressions_slow_dispatch.sh` )

dimitri-tenstorrent committed:
#8068: Remove many spurious workflows
 - Make the docs workflow callable from all commits
 - Make the job dependent onto fd-tests and sd-tests
 - Do not gate on tests success and use build machine so that the performance is more acceptable
 - Add a wrapper workflow to use build artifacts

dimitri-tenstorrent commented on .github/workflows/all-post-commit-workflows.yaml:
```diff
@@ -27,6 +33,10 @@ jobs:
     uses: ./.github/workflows/eager-package-main.yaml
     secrets: inherit
+  build-docs:
+    needs: fd-unit-tests
```
See if you can make it depend on both sets of tests `sd-unit-tests` and `fd-unit-tests`

dimitri-tenstorrent commented on .github/workflows/all-post-commit-workflows.yaml:
```diff
@@ -6,6 +6,12 @@ on:
   push:
     branches: ["main", "fd-2/main"]
 
+permissions:
```
Are these permissions ok to have on `Post commit` workflow?

tt-rkim replied:
I think pages is unavoidable...

We do need `id-token` unfortunately: https://github.com/actions/deploy-pages?tab=readme-ov-file#usage

tt-rkim commented on .github/workflows/docs-latest-public.yaml:
```diff
@@ -1,9 +1,7 @@
 name: Docs - Build latest docs and deploy to GitHub pages on main
 
 on:
-  push:
```
Hmm... I think we want to eventually use build artifacts for the docs here too.

I think for now this ok so we don't make this PR bigger than it needs to be. But we should probably file an issue to make sure the above is work is captured for the future

tt-rkim commented on .github/workflows/docs-latest-public.yaml:
```diff
@@ -1,9 +1,7 @@
 on:
-  push:
-    branches: ["main"]
-  workflow_dispatch:
```
Also why are these triggers gone? I think only the push trigger should be gone. `workflow_dispatch` is useful for those who want to build the docs on branch

dimitri-tenstorrent replied:
Good point!

tt-rkim commented:
I would recommend running both post commit and the docs workflows separately and ensuring they work

TT-billteng commented on tests/scripts/run_pre_post_commit_regressions_fast_dispatch.sh:
```diff
@@ -24,5 +24,3 @@ export PYTHONPATH=$TT_METAL_HOME
 ./tests/scripts/run_cpp_unit_tests.sh
 
-# build docs
-./tests/scripts/run_build_docs.sh
```
is this script called anywhere now?

dimitri-tenstorrent replied:
I am trying to call it from the docs workflow but it does not seem to work without a lot of other modifications.

TT-billteng replied:
I want to still give a way for devs to build the docs locally, is this the only way now?

dimitri-tenstorrent replied:
I think they can still run `./tests/scripts/run_build_docs.sh` to build docs. We do not have a tox step for docs - there are too many C++ dependencies to manage well in Tox. 

One thing I do want to do if I can is to make sure that locally and on CI the same script is executed:

i.e. locally devs call `./tests/scripts/run_build_docs.sh` and CI calls `./tests/scripts/run_build_docs.sh`

TT-billteng replied:
yes good idea, otherwise it may be broken if we don't use it in CI

dimitri-tenstorrent commented:
Build Docs - https://github.com/tenstorrent/tt-metal/actions/runs/8941441478
All Commits - https://github.com/tenstorrent/tt-metal/actions/runs/8941437613

TT-billteng commented on .github/workflows/all-post-commit-workflows.yaml:
```diff
@@ -27,6 +33,10 @@ jobs:
     uses: ./.github/workflows/eager-package-main.yaml
     secrets: inherit
+  build-docs:
+    needs: [fd-unit-tests, sd-unit-tests]
```
do you need to wait for these steps to finish? Given that you still need to compile right now on GH runners, the post-commit time is now quite a bit longer (1h 24min in your run)

dimitri-tenstorrent replied:
I can revert not to block it on the successful tests, but then we would be publishing the docs even if the tests are failing. I'll make the change to not gate on the tests.

TT-billteng replied:
think we can keep this check after it uses the build artifact

TT-billteng reviewed (changes requested):
let's not submit this until the post-commit time is back to normal; starting docs step after tests increases post-commit time by quite a bit

dimitri-tenstorrent commented:
> let's not submit this until the post-commit time is back to normal; starting docs step after tests increases post-commit time by quite a bit

Many many thanks for the review!

I have updated to use the build runner for docs and removed the gate in all_commit tests on the unit-tests.
Here are the successful runs of all-commit and build-docs workflows:
- https://github.com/tenstorrent/tt-metal/actions/runs/8950827437
- https://github.com/tenstorrent/tt-metal/actions/runs/8950828950

You can see that the build of docs take 6 mins.

TT-billteng reviewed (approved):
Looks so much cleaner, thanks !

dimitri-tenstorrent commented:
I am going to test the standalone workflow after the merge since it is a new one and it does not yet appear in the list of available workflows.
